### PR TITLE
Add tests to findAndModify on `$exists` operator

### LINKED ISSUE #1098: `upsert`: the response from FerretDB is not the same as from proxy
FerretDB version (commit) to reproduce the problem: `0c96f230f5359602d99ac440b6b66a9c91ff539e`
OS: MacOS
FerretDB is run via `task run`

## How to reproduce
Perform `findAndModify` with `upsert`, for example:
```
db.hello.findAndModify({ query: { _id: 12345}, update: {"$set": {"v": 43}}, upsert: true})
```

## Expected result
The responses from proxy and FerretDB are equal

## Actual result
The responses are different:
```2022-09-01T08:31:08.727+0200	DEBUG	// 127.0.0.1:62708 -> 127.0.0.1:27017 	clientconn/conn.go:437	Response header: length:   127, id:    7, response_to:   45, opcode: OP_MSG
2022-09-01T08:31:08.727+0200	DEBUG	// 127.0.0.1:62708 -> 127.0.0.1:27017 	clientconn/conn.go:438	Response message:
{
  "Checksum": 0,
  "FlagBits": 0,
  "Sections": [
    {
      "Document": {
        "$k": [
          "lastErrorObject",
          "value",
          "ok"
        ],
        "lastErrorObject": {
          "$k": [
            "n",
            "updatedExisting",
            "upserted"
          ],
          "n": 1,
          "updatedExisting": false,
          "upserted": 12345
        },
        "value": {
          "$k": [
            "_id",
            "v"
          ],
          "_id": 12345,
          "v": 43
        },
        "ok": {
          "$f": 1
        }
      },
      "Kind": 0
    }
  ]
}



2022-09-01T08:31:08.730+0200	DEBUG	// 127.0.0.1:62708 -> 127.0.0.1:27017 	clientconn/conn.go:437	Proxy response header: length:   106, id: 78919, response_to:   45, opcode: OP_MSG
2022-09-01T08:31:08.731+0200	DEBUG	// 127.0.0.1:62708 -> 127.0.0.1:27017 	clientconn/conn.go:438	Proxy response message:
{
  "Checksum": 0,
  "FlagBits": 0,
  "Sections": [
    {
      "Document": {
        "$k": [
          "lastErrorObject",
          "value",
          "ok"
        ],
        "lastErrorObject": {
          "$k": [
            "n",
            "updatedExisting",
            "upserted"
          ],
          "n": 1,
          "updatedExisting": false,
          "upserted": 12345
        },
        "value": null,
        "ok": {
          "$f": 1
        }
      },
      "Kind": 0
    }
  ]
}



2022-09-01T08:31:08.732+0200	DEBUG	// 127.0.0.1:62708 -> 127.0.0.1:27017 	clientconn/conn.go:223	Header diff:
--- res header
+++ proxy header
@@ -1 +1 @@
-length:   127, id:    7, response_to:   45, opcode: OP_MSG
+length:   106, id: 78919, response_to:   45, opcode: OP_MSG

Body diff:
--- res body
+++ proxy body
@@ -21,10 +21,3 @@
         },
-        "value": {
-          "$k": [
-            "_id",
-            "v"
-          ],
-          "_id": 12345,
-          "v": 43
-        },
+        "value": null,
         "ok": {
```

TODO:
- Find the current issue number mentioned in the compat tests for findAndModify, remove "skip", fix the problem and make tests pass

### LINKED ISSUE #2400: Test more `$exists` cases in `findAndModify`
### What should be done?

We have a test `TestQueryElementCompatExist` where we test query with `$exists` condition, but we don't have a similar test for `findAndModify`. 

Some cases are covered in #2385, but we should cover other possible cases (e.g. non-existent fields) too.

Possible options:
- existing and non-existent fields
- `$exists: true` and `$exists: false`
- `upsert: true` and `upsert: false`
- `$set` and `$unset`

### Definition of Done

- some `$exists` cases are covered in `TestFindAndModifyCompatUpsert`;
- all handlers updated;
- integration/compatibility tests added/updated;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #2400.
Closes #1098.

## Readiness checklist

* [ ] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/findandmodify_compat_test.go [hunk 1]
```diff
@@ -319,18 +463,94 @@ func TestFindAndModifyCompatUpsert(t *testing.T) {
...
+func TestFindAndModifyCompatUpsertSet(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]findAndModifyCompatTestCase{
+		"Upsert": {
+			command: bson.D{
+				{"query", bson.D{{"_id", "double"}}},
+				{"update", bson.D{{"$set", bson.D{{"v", 43.13}}}}},
+				{"upsert", true},
+			},
```
Comment (chilagrow) [inline integration/findandmodify_compat_test.go:480]: Covers https://github.com/FerretDB/FerretDB/issues/1098

File: integration/findandmodify_compat_test.go [hunk 2]
```diff
@@ -244,21 +403,6 @@ func TestFindAndModifyCompatUpsert(t *testing.T) {
...
-			},
-		},
-		"UpsertNew": {
-			command: bson.D{
-				{"query", bson.D{{"_id", "double"}}},
-				{"update", bson.D{{"$set", bson.D{{"v", 43.13}}}}},
-				{"upsert", true},
-				{"new", true},
-			},
-		},
```
Comment (chilagrow) [inline integration/findandmodify_compat_test.go:261]: These tests are moved to `TestFindAndModifyCompatUpsertSet`

File: integration/findandmodify_compat_test.go [hunk 3]
```diff
@@ -319,18 +463,94 @@ func TestFindAndModifyCompatUpsert(t *testing.T) {
...
+	testFindAndModifyCompat(t, testCases)
+}
+
+func TestFindAndModifyCompatUpsertUnset(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]findAndModifyCompatTestCase{
+		"NonExistentExistsTrue": {
 			command: bson.D{
 				{"query", bson.D{{"_id", bson.D{{"$exists", true}}}}},
```
Comment (rumyantseva) [inline integration/findandmodify_compat_test.go:530]: Why is the test called "non existent exists"? It uses `_id` in the query, and `_id` exists.

File: integration/findandmodify_compat_test.go [hunk 4]
```diff
@@ -319,18 +463,94 @@ func TestFindAndModifyCompatUpsert(t *testing.T) {
...
 				{"upsert", true},
-				{"update", bson.D{{"$unset", "invalid"}}},
+				{"update", bson.D{{"$unset", "v"}}},
 			},
 		},
-		"UnsetForExisting": {
+		"NonExistentExistsFalse": {
+			command: bson.D{
+				{"query", bson.D{{"_id", bson.D{{"$exists", false}}}}},
```
Comment (rumyantseva) [inline integration/findandmodify_compat_test.go:537]: Same here: `_id` field doesn't seem to be non-existent.

**Review by rumyantseva — COMMENTED** [review]
Such useful refactoring and nice tests!

Looks good to me overall, asked a couple of questions.

File: integration/findandmodify_compat_test.go (hunk 3 above)
Comment (chilagrow) [reply integration/findandmodify_compat_test.go]: I updated the test to use non existing field. Thanks for catching this :hugs:

File: integration/findandmodify_compat_test.go (hunk 4 above)
Comment (chilagrow) [reply integration/findandmodify_compat_test.go]: Also update the test to use non existing field instead of _id

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by w84thesun — APPROVED** [review]
LGTM!

**Review by noisersup — APPROVED** [review]

**Pull request merged**